sdk/core/Azure.Core/src/Shared/TrimmingAttribute.cs

```diff
@@ -0,0 +1,418 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+#if !NET5_0_OR_GREATER
```

Comment:
Just an FYI -

All the attributes here were defined in `net5.0` _except_ `RequiresDynamicCodeAttribute` which came in `net7.0`. You are fine today because you aren't using `RequiresDynamicCodeAttribute` anywhere. But if you did try to use that attribute, your `net5.0` and `net6.0` builds would fail because the attribute isn't defined in `net5/6` and you aren't adding it here.

So to be 100% correct, this should be split out into 2 groups:

```C#
#if !NET7_0_OR_GREATER
internal sealed class RequiresDynamicCodeAttribute : Attribute
{
}
#endif

#if !NET5_0_OR_GREATER
// rest of the code in this file
#endif
```